opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -868,14 +1007,27 @@ namespace cpgrid
                 // Computations in this loop could be speeded up
                 // by doing more of them simultaneously.
                 int output_face = face_to_output_face[face];
-                IndirectArray<point_t> face_pts(points, fn + fp[output_face], fn + fp[output_face+1]);
-                point_t avg = average(face_pts);
-                point_t centroid = polygonCentroid(face_pts, avg);
-                point_t normal = polygonNormal(face_pts, centroid);
-                double area = polygonArea(face_pts, centroid);
-                face_normals.push_back(normal);
-                face_centroids.push_back(centroid);
-                face_areas.push_back(area);
+                if (output_face == cpgrid::NNCFace) {
+                    // NNC faces are purely topological constructs,
+                    // and do not have any embedded geometry.
+                    // However, since the ewoms code will multiply and
+                    // divide by the face area even if not necessary
+                    // for the cell-centered FV discretization (because
+                    // it wants to deal with velocities rather than fluxes),
+                    // we have to set the areas to 1 to avoid trouble.
+                    face_normals.push_back({-1e100, -1e100, -1e100});
+                    face_centroids.push_back({-1e100, -1e100, -1e100});
+                    face_areas.push_back(1.0);
```

Comment:
I have to reiterate: Are you sure that these values will work? 

I have a feeling that they will break code at [ecltransmissibility.hh#L212-312](/OPM/ewoms/blob/2e1a6f34ca7a58a1aa1b17ec2a5cde36c2c1f38a/ebos/ecltransmissibility.hh#L212-L312). I would guess that this breaks PINCH for cases without NNC.